expression/builtin_time.go

```diff
@@ -2707,6 +2719,11 @@ func (c *addDateFunctionClass) getFunction(ctx sessionctx.Context, args []Expres
 			baseBuiltinFunc:      bf,
 			baseDateArithmitical: newDateArighmeticalUtil(),
 		}
+	case dateEvalTp == types.ETInt && intervalEvalTp == types.ETDecimal:
```

Comment:
Could you extract a function to check if we should use Decimal math, there are some redundant code, and it's easy to make mistake to write compose equal condition.